src/Manager.js

```diff
@@ -2,50 +2,38 @@
 import * as React from 'react';
 import createContext, { type Context } from 'create-react-context';
 
-export const ManagerContext: Context<{
-  setReferenceNode?: (?HTMLElement) => void,
-  referenceNode?: ?HTMLElement,
-}> = createContext({ setReferenceNode: undefined, referenceNode: undefined });
+export const ManagerReferenceNodeContext: Context<?HTMLElement> = createContext();
+export const ManagerReferenceNodeSetterContext: Context<(?HTMLElement) => void> = createContext();
 
 export type ManagerProps = {
   children: React.Node,
 };
-type ManagerState = {
-  context: {
-    setReferenceNode?: (?HTMLElement) => void,
-    referenceNode?: ?HTMLElement,
-  },
-};
 
-export default class Manager extends React.Component<
-  ManagerProps,
-  ManagerState
-> {
-  constructor() {
-    super();
-    this.state = {
-      context: {
-        setReferenceNode: this.setReferenceNode,
-        referenceNode: undefined,
-      },
-    };
-  }
+export default class Manager extends React.Component<ManagerProps> {
+  referenceNode: ?HTMLElement;
 
-  setReferenceNode = (referenceNode: ?HTMLElement) => {
-    if (!referenceNode || this.state.context.referenceNode === referenceNode) {
+  setReferenceNode = (newReferenceNode: ?HTMLElement) => {
+    if (!newReferenceNode || this.referenceNode === newReferenceNode) {
```

Comment:
Imho here we can go with 

```ts
if (this.referenceNode !== newReferenceNode) {
  this.referenceNode = newReferenceNode;
  this.forceUpdate();
} else {
  this.referenceNode = newReferenceNode;
}
```

Then we don't need to clear the  `referenceNode ` in `componentWillUnmount` and  Manager component will always have current value of reference node

Reply:
I updated to use this code, but would this recreate the issue mentioned in this comment in the original code at all? 

https://github.com/FezVrasta/react-popper/pull/266/files#r251516842

I don't think so because the problem implied there was `context` forcing a re-render due to a complex object being passed down to children, but want to double check.

Reply:
Exactly 👍